components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML2BearerGrantHandlerTest.java

```diff
@@ -0,0 +1,456 @@
+        Assert.assertFalse(saml2BearerGrantHandler.validateGrant(tokReqMsgCtx));
+
+        property2.setValue("notLocal");
+        Assert.assertFalse(saml2BearerGrantHandler.validateGrant(tokReqMsgCtx));
+
+        when(identityProviderManager
+                .getIdPByAuthenticatorPropertyValue(anyString(), anyString(), anyString(), anyString(), anyBoolean()))
+                .thenReturn(null);
+        Assert.assertFalse(saml2BearerGrantHandler.validateGrant(tokReqMsgCtx));
+
+        when(identityProviderManager
+                .getIdPByAuthenticatorPropertyValue("IdPEntityId", "localhost", "carbon.super", "samlsso", false))
+                .thenReturn(null);
+        Assert.assertFalse(saml2BearerGrantHandler.validateGrant(tokReqMsgCtx));
+
+        when(identityProviderManager
+                .getIdPByAuthenticatorPropertyValue("IdPEntityId", "localhost", "carbon.super", "SAMLSSOAuthenticator", false))
+                .thenReturn(null);
+        Assert.assertFalse(saml2BearerGrantHandler.validateGrant(tokReqMsgCtx));
+
+        property.setName("TokenEPUrl");
+        property.setValue("notLocal");
+        Assert.assertFalse(saml2BearerGrantHandler.validateGrant(tokReqMsgCtx));
+
+        when(IdentityApplicationManagementUtil.getProperty(samlConfig.getProperties(), "IdPEntityId"))
+                .thenReturn(null);
+        Assert.assertFalse(saml2BearerGrantHandler.validateGrant(tokReqMsgCtx));
+
+        when(IdentityApplicationManagementUtil.getProperty(oauthConfig.getProperties(), "OAuth2TokenEPUrl"))
+                .thenReturn(null);
+        Assert.assertFalse(saml2BearerGrantHandler.validateGrant(tokReqMsgCtx));
+
+        identityPro.setAlias("");
+        Assert.assertFalse(saml2BearerGrantHandler.validateGrant(tokReqMsgCtx));
+
+    }
+
+    @Test
+    public void testIssueRefreshToken() throws Exception {
+
+        when(oAuthServerConfiguration.getValueForIsRefreshTokenAllowed(OAuthConstants.OAUTH_SAML2_BEARER_METHOD)).thenReturn(true);
+        Assert.assertTrue(saml2BearerGrantHandler.issueRefreshToken());
+    }
+
+    @Test
+    public void testSetUser() throws Exception {
+
+        oAuth2AccessTokenReqDTO = new OAuth2AccessTokenReqDTO();
+        saml2BearerGrantHandler = new SAML2BearerGrantHandler();
+        tokReqMsgCtx = new OAuthTokenReqMessageContext(oAuth2AccessTokenReqDTO);
+        tokReqMsgCtx.setTenantID(-1234);
+        when(oAuthServerConfiguration.getSaml2BearerTokenUserType()).thenReturn(OAuthConstants.UserType.FEDERATED_USER_DOMAIN_PREFIX);
+        saml2BearerGrantHandler.setUser(tokReqMsgCtx,identityProvider,assertion,TestConstants.CARBON_TENANT_DOMAIN);
+        Assert.assertEquals(tokReqMsgCtx.getAuthorizedUser().getUserName(),assertion.getSubject().getNameID().getValue());
+
+        when(oAuthServerConfiguration.getSaml2BearerTokenUserType()).thenReturn(OAuthConstants.UserType.LOCAL_USER_TYPE);
+        ServiceProvider serviceProvider = new ServiceProvider();
```

Comment:
How about having a private method to create the Service Provider?
I saw you are creating SP in few places. Maybe you could reuse the same method